### PR TITLE
fixed opening link in WSL 

### PR DESCRIPTION
**Pull request by Fr4nk1inCs**
This PR fixes some WSL-related issue I have encountered.

## Link Format
First, Windows on my laptop cannot recognize links like `127.0.0.1:80`. For example, if I run command `cmd.exe /c start "" '127.0.0.1:80'`, the following popup shows.

image

However, if I use the link as `http://127.0.0.1:80`, Windows would open a new tab in the browser successfully. So I added `http://` prefix to host link in `find_host()`.

## Open Command in WSL
If I'm in a WSL directory, the command `cmd.exe` would start extremely slow (This would also affect the open command `cmd.exe /c start "" {url}`). For example, the screenshot below shows how slow it is to use `cmd.exe` in a WSL directory (14 seconds in `~` and 0.17 seconds in `/mnt/c`).

image

To solve this, we can use `powershell.exe -Command start "" {url}` or `explorer.exe {url}` instead. I chose `explorer.exe`.

### CODE REVIEW

**Comment by chomosuke** [comment]
Thank you very much for the PR. I'm currently very busy on the weekdays but I'll take a look at it during the weekend.

File: lua/typst-preview/commands.lua [hunk 1]
```diff
@@ -14,13 +14,13 @@ elseif
...
 then
-  open_cmd = 'cmd.exe /c start ""'
+  open_cmd = 'explorer.exe'
 else
   open_cmd = 'xdg-open'
 end
 
 local function visit(link)
-  vim.fn.jobstart(string.format('%s %s', open_cmd, link), {
+  vim.fn.jobstart(string.format('%s http://%s', open_cmd, link), {
```
Comment (chomosuke) [inline lua/typst-preview/commands.lua:23]: It'd be nice to debug log here, something like
```lua
  local cmd = string.format('%s http://%s', open_cmd, link)
  utils.debug(cmd)
  vim.fn.jobstart(cmd, {
```

**Review by chomosuke — CHANGES_REQUESTED** [review]
Currently, the first call to `:TypstPreview` does not call `visit()`. This is because by default, typst-preview binary will open the link to preview. This means that, with your changes, it'll require two call to `:TypstPreview` in WSL to open the preview in browser. I'd appreciate it if you can make the changes to fix this.

You will need to:
- Modify `vim.loop.spawn` on server.lua line 31 to pass in arguments to stop the binary from automatically opening the preview in browser.
    - You can find the relevant documentation [here](https://enter-tainer.github.io/typst-preview/standalone.html).
- Call `visit()` sometimes after `set_link(static_host)` is called in server.lua line 107.

**Comment by Fr4nk1inCs** [comment]
> Call `visit()` sometimes after `set_link(static_host)` is called in server.lua line 107.

Would it be better to call `visit()` after setting preview link in the `set_link` function in `event.spawn()` at command.lua line 71?

```diff lua
if not previewing[bufnr] then
  utils.create_autocmds(...)
  previewing[bufnr] = true
  events.watch(bufnr, function(link)
    previewing[bufnr] = link
+   visit(previewing[bufnr])
  end)
elseif type(previewing[bufnr]) == 'string' then
  print 'Opening another fontend'
  visit(previewing[bufnr])
end
```

**Comment by chomosuke** [comment]
> > Call `visit()` sometimes after `set_link(static_host)` is called in server.lua line 107.
> 
> Would it be better to call `visit()` after setting preview link in the `set_link` function in `event.spawn()` at command.lua line 71?
> 
> ```diff
> if not previewing[bufnr] then
>   utils.create_autocmds(...)
>   previewing[bufnr] = true
>   events.watch(bufnr, function(link)
>     previewing[bufnr] = link
> +   visit(previewing[bufnr])
>   end)
> elseif type(previewing[bufnr]) == 'string' then
>   print 'Opening another fontend'
>   visit(previewing[bufnr])
> end
> ```

I'd be kinda ok with that, but maybe rename `set_link` to something like `callback`since it's no longer just setting the link now.

**Comment by Fr4nk1inCs** [comment]
I'll stick with the original method, i.e., call `visit()` in server.lua. But I think I need to move the definition of `visit()` somewhere else?

**Comment by chomosuke** [comment]
Moving it to server.lua would be wise but you'd need to make to public so command.lua can still call it, probably best to move it to utils.lua

**Comment by Fr4nk1inCs** [comment]
It would be a circular require if move the definition to utils.lua, since `open_cmd` is determined using `fetch.lua` but `fetch.lua` depends on `utils.lua`

**Comment by chomosuke** [comment]
I see, in that case I'm happy for you to move it to server.lua. However, I still hold the opinion that it belongs in utils. So my preference would be to move both `visit` and `is_windows` etc to utils.lua. I do recognize that that will be a much larger change so I'd leave that decision upto you.

**Comment by Fr4nk1inCs** [comment]
That makes sense, I would move them to utils

**Comment by Fr4nk1inCs** [comment]
check this out. I also changed the open command in WSL to `/mnt/c/Windows/explorer.exe` since some people would disable Windows PATH in WSL.

File: lua/typst-preview/commands.lua [hunk 2]
```diff
@@ -2,34 +2,12 @@ local events = require 'typst-preview.events'
 local fetch = require 'typst-preview.fetch'
 local utils = require 'typst-preview.utils'
 local init = require 'typst-preview.init'
+local server = require 'typst-preview.server'
```
Comment (chomosuke) [inline lua/typst-preview/commands.lua]: Looks like this import is unused, did you forget to remove it?

**Review by chomosuke — APPROVED** [review]
All look good to me, thank you very much for the work!

**Pull request merged**